### PR TITLE
feat: Add support for express icons

### PR DESCRIPTION
**Pull request by nickschaap**
## Description

This PR modifies the build process for the icons-workflow package to also process express icons and then in a post build step uses Node conditional exports to map exports to a particular icon based off the current export conditions.

## Related issue(s)

- #2845

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by nickschaap** [comment]
It looks like there is a discrepancy in the naming conventions for the `<sp-icon-fullscreen>` and `<sp-icon-fullscreen-exit>` icons. In the Spectrum set the SVG is named `FullScreen.svg` and in the Express set is named `Fullscreen.svg`

**Comment by Westbrook** [comment]
Closing as stale. Happy to work with you or the team in yall generating an updated and complete version of this whenever you're ready!

**Comment by AndreiBaicu26** [comment]
Coming back here after some reasearch
TL;DR I still could not make this work with conditional imports. 

As far as i understand from the [docs](https://nodejs.org/api/packages.html#conditional-exports) and what i read online, NodeJs only accepts those options as conditional imports "node-addons", "node"... and are only used for deciding how to import a file depending on the environment. 

One more issue I encountered was with the naming of the icons. Since we would want a component to only use the format `<sp-my-icon>`, typescript would not allow 2 different files to create the same definition (one in spectrum icons and one in express icons): 

```ts
declare global {
    interface HTMLElementTagNameMap {
        `sp-my-icon': MyIcon;
    }
}
```

I cound not really find another way without maybe linking somehow the icon imports and the theme itself and using dynamic imports :/ 

If you have any other tips or leads that I could follow please do let me know :D
